src/tmpfiles/tmpfiles.c

```diff
@@ -2913,16 +3059,40 @@ static int parse_line(
 
         if (!empty_or_dash(age)) {
                 const char *a = age;
+                _cleanup_free_ char *seconds = NULL, *age_by = NULL;
 
                 if (*a == '~') {
                         i.keep_first_level = true;
                         a++;
                 }
 
-                r = parse_sec(a, &i.age);
+                /* Format: "age-by:age"; where age-by is "[abcm]+". */
+                if (!strchr(a, ':')) {
+                        seconds = strdup(a);
+                        if (!seconds)
+                                return log_oom();
```

Comment:
sounds unnecessary the copy the string here, just do `parse_sec(seconds ?: a, …)` further down, i.e. use `seconds` if set, and if not, use a, as before